From f61219a9890b198985da91c0f8aa62bc1c6345c5 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Mon, 16 Mar 2020 22:10:02 +0800
Subject: [PATCH 086/116] arm64: allwinner: a64: dts: add STK3311 light sensor
 to PinePhone

PinePhone has a STK3311 (or its variant) light/proximity sensor.

Add device tree node for it.

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone.dtsi | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone.dtsi b/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone.dtsi
index b2241ba987..1e97f03c78 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone.dtsi
+++ b/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone.dtsi
@@ -170,6 +170,14 @@
 			       "-1", "0", "0",
 			       "0", "0", "-1";
 	};
+
+	/* Light/Proximity sensor */
+	stk3311@48 {
+		compatible = "sensortek,stk3311";
+		reg = <0x48>;
+		interrupt-parent = <&pio>;
+		interrupts = <1 0 IRQ_TYPE_LEVEL_HIGH>; /* PB0 */
+	};
 };
 
 /* Connected to pogo pins (external spring based pinheader for user addons) */
-- 
2.21.0

